### ISSUE TITLE
Error running FloorGenerator

### ISSUE THREAD

**Issue by MahdiPishguy** [vitusortner/floor#174]
**TIP:**
i don't change any database **name**, **fields**, **tables** and i don't add any table, i run only this command :

**flutter pub run build_runner build**

> [SEVERE] floor_generator:floor_generator on lib/backend/database/app_database.dart:
> Error running FloorGenerator
> NoSuchMethodError: The getter 'name' was called on null.
> Receiver: null
> Tried calling: name

```dart
import 'dart:async';
import 'package:barcode_scanner/backend/database/entities/barcodes.dart';
import 'package:barcode_scanner/backend/database/entities/entities.dart';
import 'package:barcode_scanner/backend/models/models.dart';
import 'package:floor/floor.dart';
import 'package:sqflite/sqflite.dart' as sqflite;
import 'package:path/path.dart';
part 'app_database.g.dart';

@Database(
  version: 1,
  entities: [SessionsEntity,UserEntity,BarcodesEntity],
)
abstract class AppDatabase extends FloorDatabase {
  UserDao get userDao;

  SessionsDao get sessions;

  BarcodesDao get barcodes;
}

```

**Comment by vitusortner** [comment]
What do your DAOs looks like?

**Comment by MahdiPishguy** [comment]
@vitusortner 

```dart
@dao
abstract class UserDao{
  @Query('SELECT * FROM user LIMIT 1')
  Future<UserEntity> getUerInfo();

  @insert
  Future<void> saveUserInfo(UserEntity user);

  @update
  Future<int> updateUserInfo(UserEntity user);
}



@dao
abstract class BarcodesDao {
  @Query('SELECT * FROM barcodes')
  Future<List<BarcodesEntity>> getAllBarcodes();

  @Query('SELECT * FROM barcodes WHERE id = :id')
  Future<BarcodesEntity> getBarcodeById(int id);

  @Query('SELECT * FROM barcodes WHERE barcode = :barcode')
  Future<BarcodesEntity> getBarcodeByBarcodeData(String barcode);

  @insert
  Future<void> createNewBarcodeData(BarcodesEntity barcodesEntity);
}




@dao
abstract class SessionsDao{
  @Query('SELECT * FROM sessions  ORDER BY id DESC')
  Future<List> getAllSessionsFuture();

  @Query('SELECT * FROM sessions  ORDER BY id DESC')
  Stream<List> getAllSessionsStream();

  @insert
  Future<void> createNewSession(SessionsEntity sessionsEntity);
}
```

**Comment by MahdiPishguy** [comment]
i found problem

that was:

`Future<List> getAllSessionsFuture();`

**Comment by JakubNeukirch** [comment]
I had the same problem, for future devs having this problem: generally it is about not specified type so when you dont want some dao method return anything you should use `Future<void>` instead of `Future`, for lists the same, `Future<List>` is not enough, it must be `Future<List<YourType>>`

**Comment by alectogeek** [comment]
Having the same problem with Future<List<dynamic>>. Looks like I must specify the return data type and can't get it as dynamic and parse it later

**Comment by vitusortner** [comment]
As more people seem to run into this problem, let's make it clear in the README that defining the full return type is required.

**Comment by alectogeek** [comment]
I don't know why, but GitHub didn't show the full comment. Now it's ok, edited it adding spaces around 'dynamic'

**Comment by alectogeek** [comment]
Btw, @vitusortner, how can I query a custom data type? Now I have 2 entities in 2 tables (let's say some entity and its metadata separately) and I need to join them and get data in DTO type

**Comment by siluvana** [comment]
Run into the same issue. For future devs: my problem was a query returning a type that I forgot to add into entities list in @Database.

**Comment by sanjeetchand** [comment]
I got the same error.

I have a database view which gets generated properly. However, when I try to add a dao method which only selects certain columns instead of the whole view, I get this error.

I could not find any way to select only a few columns in the documentation so I created another class with just the columns that I need and use that as the Future<ClassName> in my dao select, but it gives this error.

Am I missing something here?
